#! /bin/sh -e
## 05_FTBFS_on_GNUHurd_and_GNUkBSD by Aníbal Monsalve Salazar
##
## DP: FTBFS on GNU/Hurd (and GNU/k*BSD)
## DP: [FIXME: we shouldn't be editing configure directly]

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
diff -urNad lynx-2.8.5/aclocal.m4~ lynx-2.8.5/aclocal.m4
--- lynx-2.8.5/aclocal.m4~	2004-01-28 20:30:38.000000000 +0100
+++ lynx-2.8.5/aclocal.m4	2004-03-14 22:02:27.000000000 +0100
@@ -4232,7 +4232,7 @@
 irix6.*) #(vi
 	CPPFLAGS="$CPPFLAGS -D_SGI_SOURCE"
 	;;
-linux*) #(vi
+linux*-gnu|gnu*|k*bsd*-gnu) #(vi
 	CF_GNU_SOURCE
 	;;
 mirbsd*) #(vi
